services/web/client/source/class/qxapp/components/workbench/servicesCatalogue/ServicesCatalogue.js

```diff
@@ -125,7 +127,7 @@ qx.Class.define("qxapp.components.workbench.servicesCatalogue.ServicesCatalogue"
       let selection = this.__list.getSelection()[0].getLabel();
       for (let i = 0; i < this.__allServices.length; i++) {
         if (selection === this.__allServices[i].label) {
-          this.fireDataEvent("AddService", [this.__allServices[i], this.__contextPort]);
+          this.fireDataEvent("AddService", [this.__allServices[i], this.__contextNodeId, this.__contextPort]);
```

Comment:
returning 'thing' positional ... for one item it is great, for 2 its ok ... for 3 and more ... hmmm how about returning a map?

Reply:
You are right, actually I need to change this in many places.  I will create a PR for that

Reply:
seperate PR ?

Reply:
Correct, so that I fix the same mistake all around the code